agent_api/src/main/java/dev/aikido/agent_api/thread_cache/ThreadCacheObject.java

```diff
@@ -72,8 +73,13 @@ public boolean isBypassedIP(String ip) {
      * Check if the IP is blocked (e.g. Geo IP Restrictions)
      */
     public BlockedResult isIpBlocked(String ip) {
-        for (BlockedIpEntry entry: blockedIps) {
-            if (entry.blocklist.matches(ip)) {
+        for (IPListEntry entry: allowedIps) {
+            if (!entry.ipList.matches(ip)) {
+                return new BlockedResult(true, entry.description);
```

Comment:
If there are multiple allowlists, the ip has to be included only in one, not in all.